dask/dataframe/io/parquet/core.py

```diff
@@ -196,8 +197,10 @@ def read_parquet(
         metadata, if present). Provide a single field name instead of a list to
         read in the data as a Series.
     filters : Union[List[Tuple[str, str, Any]], List[List[Tuple[str, str, Any]]]]
-        List of filters to apply, like ``[[('x', '=', 0), ...], ...]``. This
-        implements partition-level (hive) filtering only, i.e., to prevent the
+        List of filters to apply, like ``[[('x', '=', 0), ...], ...]``. Using this
+        argument will NOT result in row-wise filtering of the final partitions
+        unless `engine="pyarrow-dataset"` is also specified.  For other engines,
```

Comment:
Will there always be a distinction between "pyarrow-dataset" and "pyarrow"? Or will "pyarrow" start defaulting to "pyarrow-dataset" in the future and a new argument will be introduced that refers to old-pyarrow? Maybe we should introduce that new argument now so that people don't get surprised in the furure?

Reply:
Thanks for bringing this up - Right now, using `engine="pyarrow"` defaults to `"pyarrow-legacy"` (which I thik is the option you are asking about).  In the future (after we are confident with performance), `engine="pyarrow"` will default to `"pyarrow-dataset"`.  When this happens, we will need to change the docstrings like the one above to specify that row-wise filtering will **not** be performed if the user specifies `engine="pyarrow-legacy"` or `engine="fastparquet"`.

Reply:
You could spell this out explicitly: the options are fastparquet|arrow-legacy|arrow-dataset; and for now "arrow" means legacy, but this is expected to change.